src/Compilers/CSharp/Portable/Declarations/DeclarationTable.cs

```diff
@@ -150,7 +150,7 @@ internal RootNamespaceLocationComparer(CSharpCompilation compilation)
 
             public int Compare(SingleNamespaceDeclaration x, SingleNamespaceDeclaration y)
             {
-                return _compilation.CompareSourceLocations(x.Location, y.Location);
+                return _compilation.CompareSourceLocations(x.SyntaxReference, y.SyntaxReference);
```

Comment:
sorry, how does thsi cahnge anything for C#?  It looks like CompareSourceLocations does the exact same thing with Locations.

Reply:
:arrow_right: The `Location` property allocates a `SourceLocation` object each time it is accessed.

Reply:
Oh ick.  Ok, your change should def be better.  However, given how often locations are accessed (including from declarations by symbols), might we not want to also just cache this Location object?  Seems like that would also suitably help, and also benefit all the other callers that use .Location.

Reply:
I'm concerned the memory overhead by widespread caching of these objects could be problematic in other ways. This seems like the safe option for now.

Reply:
It shouldn't be.  We've added information to declarations very cheaply in the past.  It's worth noting that they only correspond to the namespaces+sourcetypes in your system.  So there are very few of them, compared to many other types of instances in our system.  On top of that, they're *heavily* shared in the most important typing paths.  All but the declarations for your file will be reused on each edit, making the allocations only usually a handful per edit per file.